tests/Behat/Context/FeatureContext.php

```diff
@@ -240,19 +240,27 @@ public function stepIFillInForForBlock($value, $name, $blockNumber)
     public function theFieldForBlockShouldContain($field, $blockNumber, $negate, $content)
     {
         $block = $this->getSpecificBlock($blockNumber);
-        $field = $this->findFieldInBlock($block, $field);
-        $isTinyMCE = $field->getAttribute('data-editor') === 'tinyMCE';
+        $fieldElem = $this->findFieldInBlock($block, $field);
+        $isTinyMCE = $fieldElem->getAttribute('data-editor') === 'tinyMCE';
 
         if ($isTinyMCE) {
             $this->cmsContext->theHtmlFieldShouldContain(
-                $field->getAttribute('name'),
+                $fieldElem->getAttribute('name'),
                 $negate,
                 $content
             );
-        } elseif ($negate) {
-            $this->assertFieldNotContains($field, $content);
+            return;
+        }
+
+        $actual = (string) $fieldElem->getValue();
+        $regex = '/^' . preg_quote($content, '/') . '$/ui';
+
+        if ($negate) {
+            $message = sprintf('The field "%s" value is "%s", but "%s" expected.', $field, $actual, $content);
+            Assert::isTrue((bool) preg_match($regex, $actual), $message);
         } else {
-            $this->assertFieldContains($field, $content);
+            $message = sprintf('The field "%s" value is "%s", but it should not be.', $field, $actual);
+            Assert::isFalse((bool) preg_match($regex, $actual), $message);
         }
```

Comment:
`assertFieldNotContains()` and `assertFieldContains()` failed because the `$field` variable wasn't a string (so obviously this part of the code just never worked but was never used).

The new assertions mirror what `assertFieldNotContains()` and `assertFieldContains()` methods _would_ have done. I don't use those directly because those will look fo the fields on the page, not on the block - and since we already have a reference to the field on the block, it's more sensible to just check against that directly here.